httpev.ml

```diff
@@ -606,7 +606,7 @@ let handle_lwt ?(single=false) fd k =
     let pause = 2. in
     log #error "too many open files, disabling accept for %s" (Time.duration_str pause);
     Lwt_unix.sleep pause
-  | `Exn Lwt.Canceled -> log #info "canceling accept loop"; Lwt.fail Lwt.Canceled
+  | `Exn (Lwt.Canceled as exn) -> log #info "canceling accept loop"; raise exn
```

Comment:
```suggestion
  | `Exn (Lwt.Canceled as exn) -> log #info "canceling accept loop"; Lwt.reraise exn
```

The `exn` is captured by `Exn_lwt.map` which should preserve the backtrace of `exn`. Using `reraise` here will add to that backtrace rather than starting a new one.

-----------------------------------

You could also write this function directly with `Lwt.try_bind` (which makes it more obvious where `Canceled`  came from) but that's somewhat orthogonal:

```
Lwt.try_bind (fun () -> Lwt_unix.accet fd)
  (fun (fd,addr as peer) -> …)
  (function
    | Unix.Unix_error (Unix.EMFILE,_,_) -> …
    | Lwt.Canceled as exn -> …
    | exn -> …)
```

Reply:
Good catch. (Force pushed it)